File: 462-minimum-moves-to-equal-array-elements-ii/minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        // sort to find the median
        sort(nums.begin(), nums.end());

        // identify the middle element
        int median = nums[nums.size() / 2];

        // sum the absolute difference
        long long moves = 0;
        for (int &num : nums) {
            moves += abs(num - median);
        }
        return moves;
    }
};

File: 462-minimum-moves-to-equal-array-elements-ii/minimum-moves-to-equal-array-elements-ii.js
var minMoves2 = function(nums) {
    nums.sort((a, b) => a - b)
    const median = nums[Math.floor(nums.length / 2)]

    return nums.reduce((total, val) => {
        return total + Math.abs(val - median)
    }, 0)
};